types/lodash/common/object.d.ts

```diff
@@ -1951,15 +1951,15 @@ declare module "../index" {
          * _.omitBy(object, _.isNumber);
          * // => { 'b': '2' }
          */
-        omitBy<T>(object: Dictionary<T> | null | undefined, predicate?: ValueKeyIteratee<T>): Dictionary<T>;
+        omitBy<T extends object>(object: T | null | undefined, predicate: ValueKeyIteratee<T[keyof T]>): {[K in keyof T]?: T[K]};
         /**
          * @see _.omitBy
          */
         omitBy<T>(object: NumericDictionary<T> | null | undefined, predicate?: ValueKeyIteratee<T>): NumericDictionary<T>;
```

Comment:
I think we can remove the `NumericDictionary` and `Dictionary` overloads completely now. All dictionaries will now be caught by the `object` overload, since it is first, so there's no point in keeping them.

However, make sure you run the tests (npm run lint lodash) in case I'm wrong about this :)